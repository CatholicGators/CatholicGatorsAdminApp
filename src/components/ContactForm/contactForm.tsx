import React from 'react';
import MaskedInput from 'react-text-mask';

import {  
  MenuItem, 
  TextField, 
  withStyles,
  createStyles
} from '@material-ui/core';

const styles = createStyles({
  container: {
      display: 'flex',
      flexWrap: 'wrap',
    },
    textField: {
      marginLeft: 10,
      marginRight: 10,
    },
    dense: {
      marginTop: 16,
    },
    menu: {
      width: 200,
    },
});

const genders = [
  {
    value: 'Male',
    label: 'Male',
  },
  {
    value: 'Female',
    label: 'Female',
  },
];

const schools = [
  {
    value: 'UF',
    label: 'UF',
  },
  {
    value: 'SFC',
    label: 'SFC',
  },
];

class ContactForm extends React.Component {
  state = {
    firstName: '',
    lastName: '',
    gender: 'Controlled',
    phoneNumber: '',
    graduationSemester: '',
    graduationYear: '',
    school: '',
    permanentAddress: '',
    city: '',
    state: '',
    zipCode: '',
    housingComplex: '',
    parent: {
      name: '',
      phone: '',
      email: ''
    }
  };

  constructor(public props: any) {
    super(props)
  }

  handleChange = name => event => {
    this.setState({
      [name]: event.target.value,
    });
  };

  handleSubmit = (event) => {
    //Make a network call somewhere
    event.preventDefault();
 }

  render() {
    const { classes } = this.props;

    return (
      <form onSubmit={this.handleSubmit}>
        <TextField 
          id="first-name"
          label="First Name" 
          className={classes.textField}
          value={this.state.firstName}
          margin="normal"
          variant="outlined"
          required
        ></TextField>
        
        <TextField 
          id="last-name"
          label="Last Name" 
          className={classes.textField}
          value={this.state.lastName}
          margin="normal"
          variant="outlined"
          required
        ></TextField>

        <TextField
          id="gender"
          select
          label="Gender"
          className={classes.textField}
          value={this.state.gender}
          onChange={(this.handleChange('gender'))}
          margin="normal"
          variant="outlined"
          required
          >
          {genders.map(option => (
            <MenuItem key={option.value} value={option.value}>
              {option.label}
            </MenuItem>
          ))}
        </TextField>

        <TextField
          id="phone-number"
          label="Phone number"
          className={classes.textField}
          value={this.state.phoneNumber}
          onChange={this.handleChange('phoneNumber')}
          margin="normal"
          variant="outlined"
          required
          >
            <MaskedInput mask={['(', /[1-9]/, /\d/, /\d/, ')', ' ', /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, /\d/]} />
          </TextField>

          /** Here lies Graduation year ans semester */

          <TextField
          id="school"
          select
          label="School"
          className={classes.textField}
          value={this.state.gender}
          onChange={(this.handleChange('school'))}
          margin="normal"
          variant="outlined"
          required
          >
          {schools.map(option => (
            <MenuItem key={option.value} value={option.value}>
              {option.label}
            </MenuItem>
          ))}
        </TextField>

        <TextField 
          id="permanent-address"
          label="Permanent Address" 
          className={classes.textField}
          value={this.state.permanentAddress}
          margin="normal"
          variant="outlined"
          required
        ></TextField>

        <TextField 
          id="city"
          label="City" 
          className={classes.textField}
          value={this.state.city}
          margin="normal"
          variant="outlined"
          required
        ></TextField>

        <TextField 
          id="state"
          label="State" 
          className={classes.textField}
          value={this.state.state}
          margin="normal"
          variant="outlined"
          required
        ></TextField>

        <TextField 
          id="zip-code"
          label="Zip Code" 
          className={classes.textField}
          value={this.state.zipCode}
          margin="normal"
          variant="outlined"
          required
        >
          <MaskedInput mask={[ /\d/, /\d/, /\d/, /\d/, /\d/]} />
        </TextField>

        <TextField 
          id="parent-name"
          label="Parent's Name" 
          className={classes.textField}
          value={this.state.parent.name}
          margin="normal"
          variant="outlined"
        ></TextField>

        <TextField 
          id="parent-phone"
          label="Parent's Phone" 
          className={classes.textField}
          value={this.state.parent.phone}
          margin="normal"
          variant="outlined"
        >
          <MaskedInput mask={['(', /[1-9]/, /\d/, /\d/, ')', ' ', /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, /\d/]} />
        </TextField>

        <TextField 
          id="parent-email"
          label="Parent's Email" 
          className={classes.textField}
          value={this.state.parent.email}
          margin="normal"
          variant="outlined"
        ></TextField>

        <TextField 
          id="housing-complex"
          label="Name of Dorm/Complex you live in" 
          className={classes.textField}
          value={this.state.housingComplex}
          margin="normal"
          variant="outlined"
        ></TextField>
        
      </form>
    );
  }
}

export default withStyles(styles)(ContactForm);
