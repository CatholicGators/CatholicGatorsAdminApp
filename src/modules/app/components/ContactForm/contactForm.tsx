import React from 'react'
import { 
  PhoneNumberTextMask, 
  ZipCodeTextMask, 
  YearTextMask 
} from './customTextMasks' 

import {  
  MenuItem, 
  TextField, 
  withStyles,
  createStyles,
  FormControl,
  InputLabel,
  OutlinedInput
} from '@material-ui/core';

const styles = createStyles({
  container: {
      display: 'flex',
      flexWrap: 'wrap',
    },
    textField: {
      marginLeft: 10,
      marginRight: 10,
    },
    dense: {
      marginTop: 16,
    },
    menu: {
      width: 200,
    }
});

const genders = [
  {
    value: 'Male',
    label: 'Male',
  },
  {
    value: 'Female',
    label: 'Female',
  },
];

const schools = [
  {
    value: 'UF',
    label: 'UF',
  },
  {
    value: 'SFC',
    label: 'SFC',
  },
];

const semester = [
  {
    value: 'Spring',
    label: 'Spring',
  },
  {
    value: 'Summer',
    label: 'Summer',
  },
  {
    value: 'Fall',
    label: 'Fall',
  },
];

class ContactForm extends React.Component {
  private inputLabelRef: React.RefObject<HTMLDivElement>

  state = {
    firstName: '',
    lastName: '',
    gender: 'Controlled',
    phoneNumber: '',
    graduationSemester: '',
    graduationYear: '',
    school: '',
    permanentAddress: '',
    city: '',
    state: '',
    zipCode: '',
    housingComplex: '',
    parentName: '',
    parentPhone: '',
    parentEmail: '',
    labelWidth: 0,
  };

  constructor(public props: any) {
    super(props)
    this.inputLabelRef = React.createRef();
    this.focusTextInput = this.focusTextInput.bind(this);
  }

  focusTextInput() {
    console.log(this.inputLabelRef.current.offsetWidth)
    this.setState({
      labelWidth: this.inputLabelRef.current.offsetWidth
    });
  }

  handleChange = (name: string) => event => {
    this.setState({
      [name]: event.target.value,
    });
  };

  handleSubmit = (event) => {
    //Make a network call somewhere
    event.preventDefault();
 }

  render() {
    const { classes } = this.props;

    return (
      <form onSubmit={this.handleSubmit}>
        <TextField 
          id="first-name"
          label="First Name" 
          className={classes.textField}
          value={this.state.firstName}
          onChange={this.handleChange('firstName')}
          margin="normal"
          variant="outlined"
          required
        ></TextField>
        
        <TextField 
          id="last-name"
          label="Last Name" 
          className={classes.textField}
          value={this.state.lastName}
          onChange={this.handleChange('lastName')}
          margin="normal"
          variant="outlined"
          required
        ></TextField>

        <TextField
          id="gender"
          select
          label="Gender"
          className={classes.textField}
          value={this.state.gender}
          onChange={this.handleChange('gender')}
          margin="normal"
          variant="outlined"
          required
          >
          {genders.map(option => (
            <MenuItem key={option.value} value={option.value}>
              {option.label}
            </MenuItem>
          ))}
        </TextField>

        <FormControl
          id="phone-number"
          className={classes.formControl} 
          variant="outlined" 
          margin="normal"
          required
        >
          <div ref={this.inputLabelRef}>
            <InputLabel 
              htmlFor="phone-number"
            >Phone number</InputLabel>
          </div>
          <OutlinedInput
            id="phone-number"
            value={this.state.phoneNumber}
            onChange={this.handleChange('phoneNumber')}
            inputComponent={PhoneNumberTextMask as any}
            labelWidth={this.state.labelWidth}
            onClick={this.focusTextInput}
            required
          />
        </FormControl>

        {/** Here lies Graduation year ans semester */}

        <FormControl
          id="graduation-year"
          className={classes.formControl} 
          variant="outlined" 
          margin="normal"
          required
        >
          <div ref={this.inputLabelRef}>
            <InputLabel
              htmlFor="graduation-year"
            >Graduation Year</InputLabel>
          </div>
          <OutlinedInput
            id="graduation-year"
            value={this.state.graduationYear}
            onChange={this.handleChange('graduationYear')}
            inputComponent={YearTextMask as any}
            labelWidth={this.state.labelWidth}
            onClick={this.focusTextInput}
            required
          />
        </FormControl>

        <TextField
          id="graduation-semester"
          select
          label="Graduation Semester"
          className={classes.textField}
          value={this.state.graduationSemester}
          onChange={this.handleChange('graduationSemester')}
          margin="normal"
          variant="outlined"
          required
          >
          {semester.map(option => (
            <MenuItem key={option.value} value={option.value}>
              {option.label}
            </MenuItem>
          ))}
        </TextField>

        <TextField
        id="school"
        select
        label="School"
        className={classes.textField}
        value={this.state.school}
        onChange={this.handleChange('school')}
        margin="normal"
        variant="outlined"
        required
        >
          {schools.map(option => (
            <MenuItem key={option.value} value={option.value}>
              {option.label}
            </MenuItem>
          ))}
        </TextField>

        <TextField 
          id="permanent-address"
          label="Permanent Address" 
          className={classes.textField}
          value={this.state.permanentAddress}
          onChange={this.handleChange('permanentAddress')}
          margin="normal"
          variant="outlined"
          required
        ></TextField>

        <TextField 
          id="city"
          label="City" 
          className={classes.textField}
          value={this.state.city}
          onChange={this.handleChange('city')}
          margin="normal"
          variant="outlined"
          required
        ></TextField>

        <TextField 
          id="state"
          label="State" 
          className={classes.textField}
          value={this.state.state}
          onChange={this.handleChange('state')}
          margin="normal"
          variant="outlined"
          required
        ></TextField>

        <FormControl
          id="zip-code"
          className={classes.formControl} 
          variant="outlined" 
          margin="normal"
          required
        >
          <div ref={this.inputLabelRef}>
            <InputLabel
              htmlFor="zip-code"
            >Zip Code</InputLabel>
          </div>
          <OutlinedInput
            id="zip-code"
            value={this.state.zipCode}
            onChange={this.handleChange('zipCode')}
            inputComponent={ZipCodeTextMask as any}
            labelWidth={this.state.labelWidth}
            onClick={this.focusTextInput}
            required
          />
        </FormControl>

        <TextField 
          id="parent-name"
          label="Parent's Name" 
          className={classes.textField}
          value={this.state.parentName}
          onChange={this.handleChange('parentName')}
          margin="normal"
          variant="outlined"
        ></TextField>

        <FormControl
          id="parent-phone-number"
          className={classes.formControl} 
          variant="outlined" 
          margin="normal"
        >
          <div ref={this.inputLabelRef}>
            <InputLabel
              htmlFor="parent-phone-number"
            >Parent's phone number</InputLabel>
          </div>
          <OutlinedInput
            id="parent-phone-number"
            value={this.state.parentPhone}
            onChange={this.handleChange('parentPhone')}
            inputComponent={PhoneNumberTextMask as any}
            labelWidth={this.state.labelWidth}
            onClick={this.focusTextInput}
          />
        </FormControl>

        <TextField 
          id="parent-email"
          label="Parent's Email" 
          className={classes.textField}
          value={this.state.parentEmail}
          onChange={this.handleChange('parentEmail')}
          margin="normal"
          variant="outlined"
        ></TextField>

        <TextField 
          id="housing-complex"
          label="Name of Dorm/Complex you live in" 
          className={classes.textField}
          value={this.state.housingComplex}
          onChange={this.handleChange('housingComplex')}
          margin="normal"
          variant="outlined"
        ></TextField>
        
      </form>
    );
  }
}

export default withStyles(styles)(ContactForm);
