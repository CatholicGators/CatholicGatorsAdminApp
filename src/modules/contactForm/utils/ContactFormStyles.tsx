import {
Theme,
createStyles
} from '@material-ui/core'

export var styles = (theme: Theme) => createStyles({
    container: {
        display: 'flex',
        flexWrap: 'wrap',
      },
      field: {
        marginLeft: '2%',
        marginRight: '2%',
        width: '46%',
        minWidth: 250
      },
      fieldSmaller: {
        marginLeft: '2%',
        marginRight: '2%',
        width: '21%',
        minWidth: 100
      },
      fieldLarger: {
        marginLeft: '2%',
        marginRight: '2%',
        width: '96%'
      },
      margins: {
        marginLeft: 10,
        marginRight: 10
      },
      layout: {
        width: 'auto',
        marginLeft: theme.spacing.unit * 2,
        marginRight: theme.spacing.unit * 2,
        [theme.breakpoints.up(600 + theme.spacing.unit * 2 * 2)]: {
          width: 600,
          marginLeft: 'auto',
          marginRight: 'auto',
        },
      },
      paper: {
        marginTop: theme.spacing.unit * 3,
        marginBottom: theme.spacing.unit * 3,
        padding: theme.spacing.unit * 2,
        [theme.breakpoints.up(600 + theme.spacing.unit * 3 * 2)]: {
          marginTop: theme.spacing.unit * 6,
          marginBottom: theme.spacing.unit * 6,
          padding: theme.spacing.unit * 3,
        },
      },
      stepper: {
        padding: `${theme.spacing.unit * 3}px 0 ${theme.spacing.unit * 5}px`,
      },
      stepLabel: {
        [theme.breakpoints.down('xs')]: {
            display: "none"
        }
      },
      buttons: {
        display: 'flex',
        justifyContent: 'flex-end',
      },
      button: {
        marginTop: theme.spacing.unit * 3,
        marginLeft: theme.spacing.unit,
      },
      spinner: {
          marginLeft: '48%',
          marginRight: '48%',
          size: 100
      }
  });
