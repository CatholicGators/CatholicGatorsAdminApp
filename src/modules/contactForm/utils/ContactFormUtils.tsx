export var steps = ['Personal information', 'Parent\'s information', 'Your interests'];

export const filterState = (state : any) => {
  var contactForm = {
    firstName: state.firstName,
    lastName: state.lastName,
    gender: state.gender,
    email: state.email,
    phoneNumber: state.phoneNumber,
    graduationSemester: state.graduationSemester,
    graduationYear: state.graduationYear,
    school: state.school,
    permanentAddress: state.permanentAddress,
    city: state.city,
    state: state.state,
    zipCode: state.zipCode,
    housingComplex: state.housingComplex,
    parentName: state.parentName,
    parentPhone: state.parentPhone,
    parentEmail: state.parentEmail,
    eventsGreekStudents: state.eventsGreekStudents,
    eventsLatinoStudents: state.eventsLatinoStudents,
    eventsGraduateStudents: state.eventsGraduateStudents,
    receiveMonthlyNewsletter: state.receiveMonthlyNewsletter,
    registerAsParishioner: state.registerAsParishioner,
    englishBibleStudy: state.englishBibleStudy,
    spanishBibleStudy: state.spanishBibleStudy,
    freeFood: state.freeFood,
    guestSpeakers: state.guestSpeakers,
    musicMinistry: state.musicMinistry,
    socials: state.socials,
    retreats: state.retreats,
    intramuralSports: state.intramuralSports,
    proLifeClub: state.proLifeClub,
    RHC: state.RHC,
    servingAtMass: state.servingAtMass,
    teachReligiousEd: state.teachReligiousEd,
    serviceProjects: state.serviceProjects
  }

  return contactForm;
}