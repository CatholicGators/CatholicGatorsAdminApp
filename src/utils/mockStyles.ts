import { createMuiTheme } from '@material-ui/core'

export default (styles) => Object.keys(styles(createMuiTheme())).reduce((acc, curr) => ({...acc, [curr]: "test"}), {})