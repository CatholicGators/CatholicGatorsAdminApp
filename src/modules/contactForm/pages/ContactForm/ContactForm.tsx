import React, { Component } from 'react'
import { connect } from 'react-redux'

import PersonalInformation from '../../components/PersonalInformation/personalInformation'
import ParentsInformation from '../../components/ParentsInformation/parentsInformation'
import Interests from '../../components/Interests/interests'
import { submitContactForm } from '../../redux/actions/contactFormActions'
import { styles } from '../../utils/ContactFormStyles'
import { steps, filterState } from '../../utils/ContactFormUtils'
import { getContactFormInterests } from '../../redux/actions/contactFormActions'

import {
    withStyles,
    Typography,
    Paper,
    Stepper,
    Step,
    StepLabel,
    CircularProgress,
    Button
} from '@material-ui/core'
import { NewContactReq } from '../../services/contactFormService'

const initState = {
    firstName: '',
    lastName: '',
    gender: '',
    email: '',
    phoneNumber: '',
    graduationSemester: '',
    graduationYear: '',
    school: '',
    permanentAddress: '',
    city: '',
    state: '',
    zipCode: '',
    housingComplex: '',
    parentName: '',
    parentPhone: '',
    parentEmail: '',
    interests: [],
    activeStep: 0
}

export class ContactForm extends Component<any, any> {
    constructor(props) {
        super(props)
        this.state = initState
    }

    componentDidMount() {
        this.props.getInterests()
    }

    handleChange(event: any, name: string) {
        if (event.target.type === 'checkbox') {
            var index = this.state.interests.indexOf(name)
            if (index < 0) {
                this.setState(prevState => {
                    return { interests: prevState.interests.concat(name) }
                })
            } else {
                var newCheckedObjectIds = this.state.interests
                newCheckedObjectIds.splice(index)
                this.setState(() => {
                    return { interests: newCheckedObjectIds }
                })
            }
        } else {
            this.setState({
                [name]: event.target.value
            })
        }
    }

    handleNext() {
        if (this.state.activeStep === steps.length - 1) {
            this.props.submitContactForm(filterState(this.state))
        } else {
            this.setState((prevState: any) => ({
                activeStep: prevState.activeStep + 1
            }))
        }
    }

    handleBack() {
        if (this.state.activeStep > 0) {
            this.setState((prevState: any) => ({
                activeStep: prevState.activeStep - 1
            }))
        }
    }

    resetStep() {
        this.setState({
            activeStep: 0
        })
    }

    render() {
        const { classes } = this.props
        const { activeStep } = this.state

        return (
            <React.Fragment>
                <main className={classes.layout}>
                    <Paper className={classes.paper}>
                        <Typography component="h1" variant="h4" align="center">
                            Join Catholic Gators
                        </Typography>
                        <Stepper
                            activeStep={activeStep}
                            className={classes.stepper}
                        >
                            {steps.map(label => (
                                <Step key={label}>
                                    <StepLabel
                                        classes={{ label: classes.stepLabel }}
                                    >
                                        {label}
                                    </StepLabel>
                                </Step>
                            ))}
                        </Stepper>
                        <React.Fragment>
                            {getStepContent(
                                this.state,
                                this.props,
                                this.handleChange.bind(this),
                                this.handleNext.bind(this),
                                this.handleBack.bind(this),
                                this.resetStep.bind(this)
                            )}
                        </React.Fragment>
                    </Paper>
                </main>
            </React.Fragment>
        )
    }
}

function getStepContent(
    state,
    props,
    handleChange,
    handleNext,
    handleBack,
    resetStep
) {
    if (props.loading) {
        return (
            <CircularProgress id="spinner" className={props.classes.spinner} />
        )
    } else if (props.success === true) {
        return (
            <React.Fragment>
                <Typography
                    component="h1"
                    variant="h6"
                    className={props.classes.margins}
                >
                    Thank you for submitting! We will be contacting you soon.
                </Typography>
                <Button
                    variant="contained"
                    color="primary"
                    href="/"
                    className={props.classes.button}
                >
                    {' '}
                    Back to form{' '}
                </Button>
            </React.Fragment>
        )
    } else if (props.success === false) {
        return (
            <React.Fragment>
                <Typography
                    component="h1"
                    variant="h6"
                    className={props.classes.margins}
                >
                    There was an error submitting. Please try again later.
                </Typography>
                <Button
                    variant="contained"
                    color="primary"
                    href="/"
                    className={props.classes.button}
                >
                    {' '}
                    Back to form{' '}
                </Button>
            </React.Fragment>
        )
    } else {
        switch (state.activeStep) {
            case 0:
                return (
                    <PersonalInformation
                        data={state}
                        handleChange={handleChange}
                        handleNext={handleNext}
                    />
                )
            case 1:
                return (
                    <ParentsInformation
                        data={state}
                        handleChange={handleChange}
                        handleNext={handleNext}
                        handleBack={handleBack}
                    />
                )
            case 2:
                return (
                    <Interests
                        interests={props.interests}
                        data={state}
                        handleChange={handleChange}
                        handleNext={handleNext}
                        handleBack={handleBack}
                    />
                )
            default:
                resetStep()
        }
    }
}

const mapDispatchToProps = dispatch => ({
    getInterests: () => dispatch(getContactFormInterests()),
    submitContactForm: (contact: NewContactReq) =>
        dispatch(submitContactForm(contact))
})

const mapStateToProps = state => ({
    interests: state.contact.interests,
    loading: state.contact.loading,
    success: state.contact.success,
    errorMessage: state.contact.errorMessage
})

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(withStyles(styles)(ContactForm))
