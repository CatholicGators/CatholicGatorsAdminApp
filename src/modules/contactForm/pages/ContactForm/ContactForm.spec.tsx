import React from 'react'
import { shallow } from 'enzyme'

import { ContactForm } from './ContactForm'
import { steps } from '../../utils/ContactFormUtils'
import mockStyles from '../../../../utils/mockStyles'
import { styles } from '../../utils/ContactFormStyles'
import toJson from 'enzyme-to-json'

describe('ContactForm', () => {
    let props, wrapper

    beforeEach(() => {
        props = {
            classes: mockStyles(styles),
            listenForUser: jest.fn(),
            submitContactForm: jest.fn()
        }
        wrapper = shallow(<ContactForm {...props} />)
    })

    it('should match snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    describe('control functions', () => {
        it('should handle next correctly', () => {
            wrapper.setState({ activeStep: 1 })
            wrapper.instance().handleNext()

            expect(wrapper.state('activeStep')).toBe(2)
        })

        it('should submit form on last next', () => {
            wrapper.setState({ activeStep: steps.length - 1 })
            wrapper.instance().handleNext()

            expect(wrapper.state('activeStep')).toBe(steps.length - 1)
            expect(props.submitContactForm).toBeCalled()
        })

        it('should change step on back', () => {
            wrapper.setState({ activeStep: 1 })
            wrapper.instance().handleBack()

            expect(wrapper.state('activeStep')).toBe(0)
        })

        it('should do nothing on back when on step 0', () => {
            wrapper.setState({ activeStep: 0 })
            wrapper.instance().handleBack()

            expect(wrapper.state('activeStep')).toBe(0)
        })

        it('should reset step on invalid step', () => {
            wrapper.setState({ activeStep: 1234 })
            wrapper.instance().resetStep()

            expect(wrapper.state('activeStep')).toBe(0)
        })
    })
})
