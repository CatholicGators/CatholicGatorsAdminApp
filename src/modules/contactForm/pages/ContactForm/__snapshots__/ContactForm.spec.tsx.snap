// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm should match snapshot 1`] = `
<Fragment>
  <main
    className="test"
  >
    <WithStyles(ForwardRef(Paper))
      className="test"
    >
      <WithStyles(ForwardRef(Typography))
        align="center"
        component="h1"
        variant="h4"
      >
        Join Catholic Gators
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Stepper))
        activeStep={0}
        className="test"
      >
        <WithStyles(ForwardRef(Step))
          key="Personal information"
        >
          <WithStyles(ForwardRef(StepLabel))
            classes={
              Object {
                "label": "test",
              }
            }
          >
            Personal information
          </WithStyles(ForwardRef(StepLabel))>
        </WithStyles(ForwardRef(Step))>
        <WithStyles(ForwardRef(Step))
          key="Parent's information"
        >
          <WithStyles(ForwardRef(StepLabel))
            classes={
              Object {
                "label": "test",
              }
            }
          >
            Parent's information
          </WithStyles(ForwardRef(StepLabel))>
        </WithStyles(ForwardRef(Step))>
        <WithStyles(ForwardRef(Step))
          key="Your interests"
        >
          <WithStyles(ForwardRef(StepLabel))
            classes={
              Object {
                "label": "test",
              }
            }
          >
            Your interests
          </WithStyles(ForwardRef(StepLabel))>
        </WithStyles(ForwardRef(Step))>
      </WithStyles(ForwardRef(Stepper))>
      <WithStyles(PersonalInformation)
        data={
          Object {
            "RHC": false,
            "activeStep": 0,
            "city": "",
            "email": "",
            "englishBibleStudy": false,
            "eventsGraduateStudents": false,
            "eventsGreekStudents": false,
            "eventsLatinoStudents": false,
            "firstName": "",
            "freeFood": false,
            "gender": "",
            "graduationSemester": "",
            "graduationYear": "",
            "guestSpeakers": false,
            "housingComplex": "",
            "intramuralSports": false,
            "lastName": "",
            "musicMinistry": false,
            "parentEmail": "",
            "parentName": "",
            "parentPhone": "",
            "permanentAddress": "",
            "phoneNumber": "",
            "proLifeClub": false,
            "receiveMonthlyNewsletter": false,
            "registerAsParishioner": false,
            "retreats": false,
            "school": "",
            "serviceProjects": false,
            "servingAtMass": false,
            "socials": false,
            "spanishBibleStudy": false,
            "state": "",
            "teachReligiousEd": false,
            "zipCode": "",
          }
        }
        handleChange={[Function]}
        handleNext={[Function]}
      />
    </WithStyles(ForwardRef(Paper))>
  </main>
</Fragment>
`;
